routers/UserSessionsRouter.py

```diff
@@ -130,5 +164,44 @@ def launch_learning(session_id: int, parameters: dict):
                 dataset_path = session.dataset_path
                 markup_path = session.data_markup_path
 
+        @router.post("/upload_dataset/{session_id}", status_code=status.HTTP_200_OK)
+        def upload_dataset(session_id: int, response: Response, file: UploadFile = File(...)):
+            try:
+                if (pathlib.Path(file.filename).suffix == '.rar') or (pathlib.Path(file.filename).suffix == '.zip'):
+                    path_to_dir = f'{os.getcwd()}\\dataset\\{session_id}'
+                    is_exist = os.path.exists(path_to_dir)
+                    if not is_exist:
+                        os.makedirs(path_to_dir)
+                    with open(f'{file.filename}', "wb") as buffer:
+                        shutil.copyfileobj(file.file, buffer)
+                    patoolib.extract_archive(archive=file.filename, outdir=path_to_dir)
+                    os.remove(file.filename)
+                    session = get_session(session_id)
+                    session.dataset_path = path_to_dir
+                    update_session(session_id, session)
+                else:
+                    response.body = f"Wrong file extension"
+                    response.status_code = status.HTTP_400_BAD_REQUEST
+            except Exception as e:
+                response.body = f"Could not upload dataset"
+                response.status_code = status.HTTP_400_BAD_REQUEST
+
+        @router.post("/upload_markup/{session_id}", status_code=status.HTTP_200_OK)
+        def upload_markup(session_id: int, response: Response, file: UploadFile = File(...)):
+            try:
+                if pathlib.Path(file.filename).suffix == '.json':
+                    path_to_markup = f'{os.getcwd()}\\dataset\\{session_id}_markup.json'
```

Comment:
markup, not dataset folder
Akso \\ slashes do not work on Linux where the app is supposed to run. Use / instead